src/stores/workspace/WorkspaceStore.ts

```diff
@@ -177,8 +178,37 @@ export default class WorkspaceStore {
 			const messages = await Promise.all(
 				messagesToLoad.map(id => this.api.messages.getMessage(id, this.attachedMessagesAC?.signal)),
 			);
+			const newStreams = [
+				...messages.map(message => message.sessionId),
+				...this.messagesStore.filterStore.filter.streams,
+			].filter((stream, index, self) => index === self.findIndex(str => str === stream));
```

Comment:
if it's filtering unique elements, i'd rather use `getArrayOfUniques` helper function